syncing/src/main/kotlin/maru/syncing/ELSyncService.kt

```diff
@@ -41,17 +46,90 @@ data class ElBlockInfo(
  * status update callback `onStatusChange` and it tries to sync the EL by `executionLayerManager` with the latest known
  * EL hash from the `beaconChain`
  */
-class ELSyncServiceImpl(
+class ELSyncService(
   private val beaconChain: BeaconChain,
-  private val leeway: UInt,
   private val executionLayerManager: ExecutionLayerManager,
   private val onStatusChange: (ELSyncStatus) -> Unit,
+  private val config: Config,
+  private val timerFactory: (String, Boolean) -> Timer = { name, isDaemon -> Timer(name, isDaemon) },
 ) : LongRunningService {
+  data class Config(
+    val pollingInterval: Duration,
+    val leeway: UInt,
+  )
+
+  private val log = LogManager.getLogger(this.javaClass)
+
+  private var poller: Timer? = null
+  private var lastELSyncTarget: ElBlockInfo? = null
+  private var currentELSyncStatus: ELSyncStatus? = null
+
+  private fun pollTask() {
+    val latestBeaconBlockHeader = beaconChain.getLatestBeaconState().latestBeaconBlockHeader
+    val latestBeaconBlockNumber = latestBeaconBlockHeader.number
+    if (latestBeaconBlockNumber == 0UL || latestBeaconBlockNumber <= config.leeway) {
```

Comment:
When latestBeaconBlockNumber is 0, we don't really know if the EL is synced or not. I don't know if it deserves a special status for this case. It will be clearer when we integrate the syncing mechanism and start testing it as a whole.